packages/nimbus/src/type-utils/omit-props.ts

```diff
@@ -15,16 +15,8 @@
  * root component to set the styles onto the `r-a-c` component using `asChild`.
  * This means that we need to allow polymorphism internally, but should not
  * allow it in the external props API since it would not work.
- *
- * @template T - The type to omit polymorphic props from
- *
- * @example
- * ```typescript
- * export type AccordionItemProps = OmitPolymorphicProps<
- *   RaDisclosureProps & HTMLChakraProps<"div">
- * > & {
- *   children: ReactNode;
- * };
- * ```
  */
-export type OmitPolymorphicProps<T> = Omit<T, "as" | "asChild">;
+export type OmitUnwantedProps<
+  T,
+  AdditionalExclusions extends string = never,
+> = Omit<T, "as" | "asChild" | "elementType" | "css" | AdditionalExclusions>;
```

Comment:
I restructured this utility type to close over props we don't want to expose internally to it, extending `Omit`. Improves maintainability so we don't have to do nested omits when we need to extend this with the component's internal ExcludedProps, a la 

```ts
ComponentProps = OmitUnwantedProps<SlotProps, ExcludedComponentProps>
```